modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/DisasterRecoveryManager.java

```diff
@@ -193,65 +195,108 @@ public CompletableFuture<Void> resetPartitions(int zoneId, int tableId) {
+            Set<String> zoneNames,
+            Set<String> nodeNames,
+            Set<Integer> partitionIds
+    ) {
         Catalog catalog = catalogManager.catalog(catalogManager.latestCatalogVersion());
 
-        return localPartitionStatesInternal(zoneName, catalog)
+        return localPartitionStatesInternal(zoneNames, nodeNames, partitionIds, catalog)
+                .thenApply(res -> checkPartitions(res, partitionIds))
                 .thenApply(res -> normalizeLocal(res, catalog));
     }
 
     /**
-     * Returns partition states for all zones' partitions in the cluster. Result is a mapping of {@link TablePartitionId} to the global
+     * Returns states of partitions in the cluster. Result is a mapping of {@link TablePartitionId} to the global
      * partition state enum value.
      *
-     * @param zoneName Zone name. {@code null} means "all zones".
+     * @param zoneNames Names specifying zones to get partition states. Empty set means "all zones".
+     * @param partitionIds IDs of partitions to get states of. Empty set means "all partitions".
      * @return Future with the mapping.
      */
-    public CompletableFuture<Map<TablePartitionId, GlobalPartitionState>> globalPartitionStates(@Nullable String zoneName) {
+    public CompletableFuture<Map<TablePartitionId, GlobalPartitionState>> globalPartitionStates(
+            Set<String> zoneNames,
+            Set<Integer> partitionIds
+    ) {
         Catalog catalog = catalogManager.catalog(catalogManager.latestCatalogVersion());
 
-        return localPartitionStatesInternal(zoneName, catalog)
+        return localPartitionStatesInternal(zoneNames, Set.of(), partitionIds, catalog)
+                .thenApply(res -> checkPartitions(res, partitionIds))
                 .thenApply(res -> normalizeLocal(res, catalog))
                 .thenApply(res -> assembleGlobal(res, catalog));
     }
 
     private CompletableFuture<Map<TablePartitionId, Map<String, LocalPartitionStateMessage>>> localPartitionStatesInternal(
-            @Nullable String zoneName, Catalog catalog
+            Set<String> zoneNames,
+            Set<String> nodeNames,
+            Set<Integer> partitionIds,
+            Catalog catalog
     ) {
-        int zoneId;
-        if (zoneName == null) {
-            zoneId = NO_ZONE_ID;
+        Set<Integer> zoneIds;
+        if (zoneNames.isEmpty()) {
+            zoneIds = Set.of();
         } else {
-            Optional<CatalogZoneDescriptor> zoneDesciptorOptional = catalog.zones().stream()
-                    .filter(catalogZoneDescriptor -> catalogZoneDescriptor.name().equals(zoneName))
-                    .findAny();
+            Set<String> uppercaseZoneNames = zoneNames.stream()
+                    .map(String::toUpperCase)
+                    .collect(toSet());
+
+            List<CatalogZoneDescriptor> zoneDescriptors = catalog.zones().stream()
```

Comment:
In previous changes, zone names were compared for equality without converting to uppercase, why is this so now?

If you try the following code in `org.apache.ignite.internal.sql.engine.ItCreateTableDdlTest`, then two queries will be executed without errors, so I think converting to uppercase is not correct.

```
sql("CREATE ZONE TEST_ZONE WITH STORAGE_PROFILES='" + DEFAULT_STORAGE_PROFILE + "'");
sql("CREATE ZONE \"test_zone\" WITH STORAGE_PROFILES='" + DEFAULT_STORAGE_PROFILE + "'");
```

Let's make the code related to searching for nodes a separate method.

If I understand correctly, then the logic is missing; if the names of the zones are not specified, then we take all the zones.